rust/kernel/sync/mod.rs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Synchronisation primitives.
+//!
+//! This module contains the kernel APIs related to synchronisation that have been ported or
+//! wrapped for usage by Rust code in the kernel and is shared by all of them.
+//!
+//! # Example
```

Comment:
I'd move the example into `mutex.rs`, i.e. so that we have a different one for condvar, spinlock, etc.

Not sure if it is best to also move each of them to `mutex_init!`, rather than have them in the module doc, hmm... I will think about it, but we can move it later.

Reply:
If we move it to `mutex.rs`, would it be visible in the rendered docs? (Given that is gets 'exported' via `pub use`.)

Reply:
There is an attribute for pub use, to render the doc in place, so yes we can rerender it as it would be in this file.